pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -281,10 +323,52 @@ func manageRouteControllerManagerClientCA_v311_00_to_latest(client coreclientv1.
 	return resourceapply.SyncConfigMap(context.Background(), client, recorder, util.KubeAPIServerNamespace, apiserverClientCA, util.RouteControllerTargetNamespace, apiserverClientCA, []metav1.OwnerReference{})
 }
 
+// groupDisabledControllers gets caps from configv1.ClusterVersion and populates
+// groupDisabled with all controllers that are known in the ClusterVersion and
+// also disabled.
+func groupDisabledControllers(clusterVersion *configv1.ClusterVersion) map[controlplanev1.OpenShiftControllerName]bool {
+	knownCaps := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.KnownCapabilities...)
+	capsEnabled := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.EnabledCapabilities...)
+
+	groupDisabled := make(map[controlplanev1.OpenShiftControllerName]bool)
+	for _, cont := range knownControllers {
```

Comment:
Given only two controllers can be disabled at the moment, it might be easier to just iterate through them and check for the capabilities. I.e.
```
var controllerCapabilities = map[controlplanev1.OpenShiftControllerName]configv1.ClusterVersionCapability{
	controlplanev1.OpenshiftBuildController:                     configv1.ClusterVersionCapabilityBuild,
	controlplanev1.OpenshiftDeploymentConfigController:          configv1.ClusterVersionCapabilityDeploymentConfig,
}
...
for cont, cap := range controllerCapabilities {
	if knownCaps.Has(cap) && !capsEnabled.Has(cap) {
		groupDisabled[cont] = true
	}
}
...
```